src/library/views.py

```diff
@@ -283,6 +292,85 @@ def post(self, request, campaign_import_id):
         return redirect('quest_manager:categories_inactive')
 
 
+@method_decorator([login_required, staff_member_required], name='dispatch')
+class ExportQuestView(View):
+    """
+    View for exporting a single quest from the current deck into the shared library.
+
+    Only the deck owner may export unless `allow_staff_export` is set to True.
+    """
+    template_name = 'library/confirm_export_quest.html'
+
+    def _require_export_permission(self, request):
+        config = SiteConfig.get()
+
+        if not (config.allow_staff_export or request.user == config.deck_owner):
+            raise PermissionDenied("Only the deck owner can export quests unless allow_staff_export is enabled.")
+
+    def get(self, request, quest_import_id):
+        self._require_export_permission(request)
+
+        quest = get_object_or_404(Quest.objects.all_including_archived(), import_id=quest_import_id)
+
+        with library_schema_context():
+            library_quest = Quest.objects.filter(import_id=quest.import_id).first()
+
+        return render(request, self.template_name, {
+            'quest': quest,
+            'library_quest': library_quest,
+        })
+
+    def post(self, request, quest_import_id):
+        self._require_export_permission(request)
+
+        quest = get_object_or_404(Quest.objects.all_including_archived(), import_id=quest_import_id)
+
+        source_schema = connection.schema_name
+
+        with library_schema_context():
+            if Quest.objects.filter(import_id=quest.import_id).exists():
+                raise PermissionDenied(f"A quest with import_id {quest.import_id} already exists in the shared library.")
+
+            # Perform export
+            export_quest_to_library(source_schema=source_schema, quest_import_id=quest.import_id)
+
+            # Get the newly exported quest
+            exported_quest = Quest.objects.get(import_id=quest.import_id)
+
+            config = SiteConfig.get()
+            recipient = config.deck_owner
+            sender = config.deck_ai
+
+            # Fallback: If sender and recipient are the same, the notification system will skip sending.
+            # Pick another staff user to act as sender if possible.
+            if sender == recipient:
+                sender = (
+                    User.objects
+                    .filter(is_active=True, is_staff=True)
+                    .exclude(id=recipient.id)
+                    .order_by('id')
+                    .first()
+                )
+
+            # Only send the notification if both sender and recipient are valid.
+            # This avoids silent failures in setups with only one user.
```

Comment:
Why would there only be one user?  A new deck has at least two, the admin and the owner, doesn't it?

Reply:
This seems like something to fix in the notification app rather than here.  Maybe create an issue for it if it's a legit problem.

Reply:
I'm confused why these are silent failures.  What is failing?  Why would you want to send a notification to yourself?

Can you add some context? Which deck is this in, the library or the local deck sharing the quest?  Can you add docstrings to these methods with details?